src/Elm/Parser/Layout.elm

```diff
@@ -37,6 +37,19 @@ nonEmptyWhiteSpaceOrComment =
 
 whiteSpaceAndComments : Parser Comments
 whiteSpaceAndComments =
+    Parser.oneOf
+        [ Parser.keyword " "
+            |> Parser.andThen (\() -> whiteSpaceAndCommentsLoop)
+        , Parser.token " "
+            |> Parser.map (\() -> Rope.empty)
+
+        -- fallback if the hacky shortcuts don't commit
+        , whiteSpaceAndCommentsLoop
```

Comment:
Could you explain the trick here? I'm not sure I get it.

Reply:
The trick was: Since a single space is by far the most common maybeLayout, we
- first check whether there is not text/numbers/... after the space using `keyword`. If there isn't, we can assume (if elm-format is used) that there will be more whitespace following that (that's obviously not a general rule but a common rule because of how common references/variables are)
- If there is, we (if elm-format is used) very likely have a single space which we can now reach and parse very fast

These are very fragile/stupid heuristics that did make a small dent. I'm happy the new thing is faster without.